### ISSUE TITLE
Latest release breaks mobile dragging

### ISSUE THREAD

**Issue by christiancruuz** [bokuweb/react-rnd#725]
My browser is:   Mobile Chrome and Mobile Safari on ios
  

Latest release breaks mobile dragging. When I reverted back to version 10.1.1 it worked.

**Comment by bokuweb** [comment]
@chrisstiles Do you have any idea?

**Comment by chrisstiles** [comment]
@bokuweb I'll look into it and see what I can find. It does like the `nodeRef` is what is breaking dragging on mobile.

**Comment by chrisstiles** [comment]
@bokuweb I was able to fix the issue preventing dragging on mobile!

The root of the issue is caused by the fact that React-Draggable is using the newer React ref syntax and expects a `RefObject` with the DOM element being the `ref.current` value. Since re-resizable uses callback refs, I had previously just created an object in the `render()` function and set the resizable `<div>` as the value. 

However, because that object was always created during a render, React-Draggable wouldn't always have the latest reference to the DOM element. Instead, I added a `resizableElement` object as an instance variable on the `<RND />` component, and update its `current` value whenever `refResizable()` is called.

Although dragging worked, **I found another bug that existed before we added the `nodeRef` prop.** When resizing on mobile, the box's position would also shift around unexpectedly. I ran the storybook with the `index.tsx` code from a previous version and found was still happening there.

This one was caused by `onResize()` method, although I'm not quite sure why it only happened on mobile. When resizing the box from certain sides (ie from the left), we have to move its position as well. It looks like the `onResize()` function was doing that but not handling all the scenarios correctly.

I fixed this issue by changing the position logic in `onResize()` to the following. The array of directions includes all the scenarios where we'd need to change the box's position while resizing.

```typescript
const newPos = { x: this.state.original.x, y: this.state.original.y };
const left = -delta.width;
const top = -delta.height;
const directions = ["top", "left", "topLeft", "bottomLeft", "topRight"];

if (directions.includes(direction)) {
  if (direction === "bottomLeft") {
    newPos.x += left;
  } else if (direction === "topRight") {
    newPos.y += top;
  } else {
    newPos.x += left;
    newPos.y += top;
  }
}

if (newPos.x !== this.draggable.state.x || newPos.y !== this.draggable.state.y) {
  this.draggable.setState(newPos);
}
```

I tested out these changes on multiple browsers, my iPhone and on Android (with Browserstack). Everything seems to be working as expected now 😃! I'll submit a PR with the changes.
